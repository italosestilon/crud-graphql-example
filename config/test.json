{
    "dbHost": "localhost",
    "dbName": "db",
    "env": "test"
}