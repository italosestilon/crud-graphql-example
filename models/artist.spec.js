import config from "config";
import Artist from "./artist";

import setupTest from "../test/setup";

import mongoose from "mongoose";

import { sanitizeObject } from "../test/helper";

const Schema = mongoose.Schema;

beforeEach(async () => {
  await setupTest();
});

afterAll(async () => {
  await disconnect();
});

it("should not save an empty artist", async () => {
  try {
    const artist = new Artist({});
    await artist.save();
  } catch (err) {
    expect(err).toMatchSnapshot();
  }
});

it("should save new artist", async () => {
  try {
    const artist = new Artist({
      name: "Lana Del Rey"
    });

    await artist.save();

    expect(sanitizeObject(artist._doc)).toMatchSnapshot();
  } catch (err) {
    expect(err).toMatchSnapshot();
  }
});

it("should retrieve an artist", async () => {
  try {
    const artist = new Artist({
      name: "Lana Del Rey"
    });

    await artist.save();

    const retrievedArtist = await Artist.findById(artist.id);

    expect(sanitizeObject(retrievedArtist._doc)).toMatchSnapshot();
  } catch (err) {
    expect(err).toMatchSnapshot();
  }
});

it("should delete an artist and all her albums", async () => {
  try {
    const artist = new Artist({
      name: "Lana Del Rey"
    });

    await artist.save();

    const retrievedArtist = await Artist.deleteOne({ id: artist.id });

    expect(sanitizeObject(retrievedArtist._doc)).toMatchSnapshot();
  } catch (err) {
    expect(err).toMatchSnapshot();
  }
});

async function disconnect() {
  try {
    await mongoose.disconnect();
  } catch (err) {
    console.log(err);
  }
}
