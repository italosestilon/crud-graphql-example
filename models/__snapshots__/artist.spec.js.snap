// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should delete an artist 1`] = `[TypeError: _artist2.default.removeById is not a function]`;

exports[`should not save an empty artist 1`] = `[ValidationError: Artist validation failed: name: Path \`name\` is required.]`;

exports[`should retrieve an artist 1`] = `
Object {
  "__v": 0,
  "_id": "ObjectId",
  "albuns": Array [],
  "name": "Lana Del Rey",
}
`;

exports[`should save new artist 1`] = `
Object {
  "__v": 0,
  "_id": "ObjectId",
  "albuns": Array [],
  "name": "Lana Del Rey",
}
`;
